### PR TITLE
WPSC: Add Debug Log Buttons

### PR DESCRIPTION
**Pull request by ockham**
Requires an up-to-date `master` branch of WPSC.

To test:
* The Debug Log card should only be visible if the 'Enable Debugging' toggle is enabled.
* SSH into your wpsandbox, `cd` to `$cache_path` (usually `wp-content/cache/`)
* Watch those cryptic hexadecimal `46ae84b26d.php` log files.
* Click the `Reset Logs` button. The current log file should shrink in size.
* Click the `Disable Logs` button. The Debug Log card should vanish, and the 'Enable Debugging' toggle will be disabled.
* Click the `Disable and Delete Logs` button. The Debug Log card should vanish, and the 'Enable Debugging' toggle will be disabled; furthermore, the current log file will be deleted.

TODO:
* `items` reducer:
  * After disabling, set the previously active log to inactive; tests!
  * Upon enabling debugging: Add new debug log to `items` (needs endpoint change); tests!
  * Add test for `_DELETE`
* 'Delete All' button
* Success notice: Change wording to 'Successfully reset' for active log

### CODE REVIEW

**Comment by ockham** [comment]
@donnchawp Can we expose cache location as a `status`? What about username/pw, is that safe to be exposed thru the API?

UI question: Does the 'Disable Debug Log' button do anything different from the 'Enable Debugging' toggle?

**Comment by donnchawp** [comment]
The cache location is already exposed as cache_path in /settings/.
The u/p is only used for the debug log, which up until the next release was simply a text file with an md5 filename.txt - that information is shown in the wp-admin/ settings page too and changes every time you create a new debug log so I'm not worrying too much about it.
It's almost as bad transmitting the cache location in the clear but since every WordPress blog with Supercache has been doing that for years I think the risk of MITM attacks is small.

You're right about the disable button. It's a convenience on the wp-admin page because the toggle is part of a form that has to be submitted by a button that's below the fold. You can probably get rid of it from Calypso.

**Comment by ockham** [comment]
> The cache location is already exposed as cache_path in /settings/.

But that's only a path, not the filename as displayed in the wp-admin Debug tab, is it?

> You're right about the disable button. It's a convenience on the wp-admin page because the toggle is part of a form that has to be submitted by a button that's below the fold. You can probably get rid of it from Calypso.

Cool. What would we do about the 'Disable and Delete Log' button? Just make it delete the log, not disable debugging?

**Comment by donnchawp** [comment]
The filename of the log file is found in "wp_cache_debug_log". It's a .php file, so just append it to the cache_path.

The "Reset Debug Log" button deletes the log file and recreates it again (with the same name and u/p) so the "Disable and Delete log" button is still needed. If you toggle debugging that leaves the log file in place for further analysis.

Maybe the UI needs further editing though. The page should probably list previous debug logs that haven't been deleted with a delete button next to each. We wouldn't need the disable and delete button then.

**Comment by ockham** [comment]
> The filename of the log file is found in "wp_cache_debug_log". It's a .php file, so just append it to the cache_path.

Thanks, managed to assemble the full path using that way!

> The "Reset Debug Log" button deletes the log file and recreates it again (with the same name and u/p) so the "Disable and Delete log" button is still needed. If you toggle debugging that leaves the log file in place for further analysis.

> Maybe the UI needs further editing though. The page should probably list previous debug logs that haven't been deleted with a delete button next to each. We wouldn't need the disable and delete button then.

Sounds good. Since in its current state, this PR gets us to feature parity with wp-admin's Debug Log buttons, I'm inclined to leave that for another iteration tho :slightly_smiling_face:

File: client/extensions/wp-super-cache/components/debug/index.jsx [hunk 1]
```diff
@@ -49,6 +59,54 @@ class DebugTab extends Component {
...
+						</FormFieldset>
+						<p>
+							{ translate(
+								'Currently logging to: %(location)s',
+								{ args: { location: cache_path + wp_cache_debug_log } }
+							) }
+						</p>
+						<p>
+							{ translate(
+								'Username and Password: %(username)s',
```
Comment (a8ci18n) [inline client/extensions/wp-super-cache/components/debug/index.jsx]: Hi! I've found a possible matching string that has already been translated **82** times:
`translate( '[%s] Your username and password' )` *ES Score: 17*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/wpsc-debug-log-buttons)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/wp-super-cache/components/debug/index.jsx [hunk 2]
```diff
@@ -95,8 +98,15 @@ class DebugTab extends Component {
 						</FormFieldset>
 						<p>
 							{ translate(
-								'Currently logging to: %(location)s',
-								{ args: { location: cache_path + wp_cache_debug_log } }
+								'Currently logging to: {{ExternalLink}}%(location)s{{/ExternalLink}}',
+								{
+									args: { location: cache_path + cacheFilename },
+									components: {
+										ExternalLink: <ExternalLink
```
Comment (donnapep) [inline client/extensions/wp-super-cache/components/debug/index.jsx]: I'm not seeing that icon that I thought the `ExternalLink` component was supposed to add. Should it be here?

Comment (ockham) [inline client/extensions/wp-super-cache/components/debug/index.jsx]: There's a dedicated prop needed to enable it. It looked bad when inlined into that sentence, so I dropped it.

**Comment by ockham** [comment]
> When I click any of the buttons, I get a "Settings saved" notification. Should those notifications be customized for these buttons?

That's because I'm [piggybacking](https://github.com/Automattic/wp-calypso/pull/16026/files#diff-82589d7f87af9eb87adbd3592a7b7ac4R33) `saveSettings` here, since I lazily didn't want to introduce new actions etc. But yeah, UX-wise, that's a bit dubious...

> I wonder if the card with the buttons should move below the Debug card? It felt a bit jarring when I turned debugging on.

Hmm, I'm copying the wp-admin UI here. The card does look a bit scary when it suddenly appears on the very top. OTOH, as @donnchawp pointed out, apparently the idea is to have those actions up there and not somewhere further down (like below the fold). Wanna weigh in on this @donnchawp?

**Comment by donnchawp** [comment]
Maybe we should bite the bullet and make an array of debug logs+username. Then at least we could have one on/off switch and display the list of log files below or above or somewhere. No need for the different actions. Each log file will have a delete button, and if debugging is active then it'll have a reset button too.

**Comment by ockham** [comment]
> Maybe we should bite the bullet and make an array of debug logs+username. Then at least we could have one on/off switch and display the list of log files below or above or somewhere. No need for the different actions. Each log file will have a delete button, and if debugging is active then it'll have a reset button too.

Okay, sounds sensible from a UX POV. Means the REST API will have to change a bit too tho...

**Comment by donnapep** [comment]
> I'm copying the wp-admin UI here.

I'll just add that we are free to do some things differently in Calypso, whether that's changing the order of things, updating the text etc. It doesn't have to be identical. Our only real constraint is that we should implement the same functionality.

File: client/extensions/wp-super-cache/components/debug/index.jsx [hunk 3]
```diff
@@ -44,13 +57,65 @@ class DebugTab extends Component {
...
+
+					{ !! wp_super_cache_debug &&
+						<div>
+							<SectionHeader label={ translate( 'Debug Logs' ) } />
+							<Card>
+								<table>
+									<thead>
+										<tr>
+											<th>{ translate( 'Filename' ) }</th>
+											<th>{ translate( 'Username/Password' ) }</th>
```
Comment (a8ci18n) [inline client/extensions/wp-super-cache/components/debug/index.jsx]: Hi! I've found a possible matching string that has already been translated **50** times:
`translate( 'Your username and password' )` *ES Score: 11*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/wpsc-debug-log-buttons)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/wp-super-cache/components/debug/index.jsx [hunk 4]
```diff
@@ -44,13 +57,65 @@ class DebugTab extends Component {
...
+					</Card>
+
+					{ !! wp_super_cache_debug &&
+						<div>
+							<SectionHeader label={ translate( 'Debug Logs' ) } />
+							<Card>
+								<table>
+									<thead>
+										<tr>
+											<th>{ translate( 'Filename' ) }</th>
```
Comment (a8ci18n) [inline client/extensions/wp-super-cache/components/debug/index.jsx]: Hi! I've found a possible matching string that has already been translated **98** times:
`translate( 'Empty filename' )` *ES Score: 14*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/wpsc-debug-log-buttons)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/wp-super-cache/components/debug/index.jsx [hunk 5]
```diff
@@ -44,13 +57,61 @@ class DebugTab extends Component {
...
+						</FormToggle>
+					</Card>
+
+					<SectionHeader label={ translate( 'Debug Logs' ) } />
+					<Card>
+						<table>
+							<thead>
+								<tr>
+									<th>{ translate( 'Filename' ) }</th>
+									<th>{ translate( 'Username/Password' ) }</th>
```
Comment (a8ci18n) [inline client/extensions/wp-super-cache/components/debug/index.jsx]: Hi! I've found a possible matching string that has already been translated **50** times:
`translate( 'Your username and password' )` *ES Score: 11*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/wpsc-debug-log-buttons)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/wp-super-cache/components/debug/index.jsx [hunk 6]
```diff
@@ -44,13 +57,61 @@ class DebugTab extends Component {
...
+							{ translate( 'Enable Debugging' ) }
+						</FormToggle>
+					</Card>
+
+					<SectionHeader label={ translate( 'Debug Logs' ) } />
+					<Card>
+						<table>
+							<thead>
+								<tr>
+									<th>{ translate( 'Filename' ) }</th>
```
Comment (a8ci18n) [inline client/extensions/wp-super-cache/components/debug/index.jsx]: Hi! I've found a possible matching string that has already been translated **98** times:
`translate( 'Empty filename' )` *ES Score: 14*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/wpsc-debug-log-buttons)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by ockham** [comment]
Closing in favor of #16282.

**Pull request closed without merging**